tests/linker/mac/LinkAnyTest.cs

```diff
@@ -21,15 +21,27 @@ public void AES ()
 		}
 
 		static bool waited;
+		static bool requestError;
+		static HttpStatusCode statusCode;
+
+
 		// http://blogs.msdn.com/b/csharpfaq/archive/2012/06/26/understanding-a-simple-async-program.aspx
 		// ref: https://bugzilla.xamarin.com/show_bug.cgi?id=7114
 		static async Task GetWebPageAsync ()
 		{
-			Task<string> getWebPageTask = new HttpClient ().GetStringAsync ("http://msdn.microsoft.com");
-			string content = await getWebPageTask;
-			waited = true;
-			bool success = !String.IsNullOrEmpty (content);
-			Assert.IsTrue (success, $"received {content.Length} bytes");
+			// do not use GetStringAsync, we are going to miss useful data, such as the resul code
+			using (var client = new HttpClient ()) {
+				HttpResponseMessage response = await client.GetAsync ("http://msdn.microsoft.com"); 
+				if(!response.IsSuccessStatusCode) {
+					requestError = true;
+					statusCode = response.StatusCode;
+				} else {
+					string content = await response.Content.ReadAsStringAsync ();
+					waited = true;
+					bool success = !String.IsNullOrEmpty (content);
```

Comment:
tiny nit: it's a little confusing to have a whole `bool success` variable that we expect to be true always. 

In the context of http request code, I would expect a variable named `success` to indicate something like `response.IsSuccessStatusCode`, but it seems like by the time we hit the `else` branch we're already in the success case :)

Otherwise, looks good to me.

Reply:
good point. I think we are just testing that async/await works. If that is the case, even the errno would be good.

@rolfbjarne do you know why do we have this tests exactly? I looked at the bugzilla issue and might just be for the async support.

Reply:
The test was added here: https://github.com/xamarin/maccore/commit/ca7dd9294d173893fbb2b635175141797261ead8, and yes, it looks like it was just to make sure async/await works.

But did you try just changing the url to `http://example.com`? That's a url we're already using in other places, and which will fail if the url goes down. It's possible http://msdn.microsoft.com will always return us a 500 code, and it doesn't feel right to ignore any such failures (eventually we'll end up ignoring legitimate failures).

Reply:
make sense to reduce the number of endpoint our tests depends on... maybe we should have them as constants (in a different PR)